schema/map_test.go

```diff
@@ -362,3 +363,49 @@ func TestMap_UnSerialize_Reversible(t *testing.T) {
 	// test reversiblity
 	assert.Equals(t, unserialized2, unserialized)
 }
+
+// Test type aliased map of type aliased string to a struct.
+type TypedStringEnumTestType string
+
+const (
+	testA TypedStringEnumTestType = "testA"
+	testB TypedStringEnumTestType = "testB"
+)
+
+func TestMap_AliasedTypes(t *testing.T) {
+	schemaForPrivateFieldStruct := schema.NewStructMappedObjectSchema[testdata.TestStructWithPrivateField](
+		"structWithPrivateField",
+		map[string]*schema.PropertySchema{
+			"field1": schema.NewPropertySchema(
+				schema.NewStringSchema(nil, nil, nil),
+				nil,
+				false,
+				nil,
+				nil,
+				nil,
+				schema.PointerTo("\"Hello world!\""),
+				nil,
+			),
+		},
+	)
+	schemaForEnum := schema.NewTypedStringEnumSchema[TypedStringEnumTestType](map[TypedStringEnumTestType]*schema.DisplayValue{
+		testA: nil,
+		testB: nil,
+	})
+	mapType := schema.NewMapSchema(
+		schemaForEnum,
+		schemaForPrivateFieldStruct,
+		nil,
+		nil,
+	)
+
+	unserializedData, err := mapType.Unserialize(map[string]any{
+		string(testA): map[string]any{
+			"field1": "test_field_value",
```

Comment:
I think part of the point is that the private `field2`  in your reference type doesn't interfere with any of the type matching logic. So would it be a stronger (or usefully complementary) test to try this with a (presumably ignored and irrelevant) `field2` value?

Reply:
Generally I'd prefer thorough testing of a new feature when it's added; but if it makes more sense to add testing in an additional layer, we can hopefully recapture the tech debt quickly. 😆

Reply:
This is just work done to test private fields. It was under-tested before, but no changes to the logic have been added for private fields.